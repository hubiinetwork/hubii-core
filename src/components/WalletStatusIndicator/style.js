import styled from 'styled-components';

export const Slanted = styled.div`
  position: relative;
  :before {
    content: '';
    position: absolute;
    top: 0;
    left: -4.43rem;
    width: 0;
    height: 0;
    border-right: 7.14rem solid transparent;
    border-top: 7.14rem solid
      ${(props) =>
    props.active ? props.theme.palette.info : props.theme.palette.secondary6};
  }
  transform: translateZ(0px); /* fix overflow when spinner icon */
  -webkit-transform: translateZ(0px); 
`;

export const Icon = styled.img`
  position: absolute;
  max-width: 1.14rem;
  max-height: 1.14rem;
  top: ${({ walletType }) => walletType === 'hardware' ? '0.43' : '0.3'}rem;;
  left: ${({ walletType }) => walletType === 'hardware' ? '0.43' : '0.2'}rem;
`;

export const TextWhite = styled.span`
  color: ${({ theme }) => theme.palette.light};
`;
