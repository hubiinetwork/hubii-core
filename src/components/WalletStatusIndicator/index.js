import React from 'react';
import PropTypes from 'prop-types';
import { Tooltip } from 'antd';

import USBConnected from '../../../public/images/usb-icon.svg';
import USBDisconnected from '../../../public/images/usb-icon-white.svg';
import Locked from '../../../public/images/lock.png';
import Unlocked from '../../../public/images/lock_open.svg';

import { Slanted, Icon } from './style';

/**
 * This component shows status of the user's API connection with wallet by showing connected flag.
 * For proper UI, give style={{overflow:'hidden}} to the parent container.
 */

const WalletStatusIndicator = ({ active, walletType }) => {
  let title;
  let icon;
  if (walletType === 'software') {
    if (active) {
      title = 'Unlocked';
      icon = Unlocked;
    } else {
      title = 'Locked';
      icon = Locked;
    }
  } else if (active) {
    title = 'Connected';
    icon = USBConnected;
  } else {
    title = 'Disconnected';
    icon = USBDisconnected;
  }
  return (
    <Slanted active={active}>
      <Tooltip
        placement="right"
        title={title}
      >
        <Icon walletType={walletType} src={icon} />
      </Tooltip>
    </Slanted>
  );
};
WalletStatusIndicator.propTypes = {
  active: PropTypes.bool.isRequired,
  walletType: PropTypes.oneOf(['software', 'hardware']),
};

export default WalletStatusIndicator;
