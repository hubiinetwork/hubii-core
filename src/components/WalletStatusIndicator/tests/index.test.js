import React from 'react';
import { shallow } from 'enzyme';

import WalletStatusIndicator from '../index';

describe('<WalletStatusIndicator />', () => {
  it('should render correctly when wallet is connected hardware', () => {
    const wrapper = shallow(
      <WalletStatusIndicator
        active
        walletType={'hardware'}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when wallet is disconnected hardware', () => {
    const wrapper = shallow(
      <WalletStatusIndicator
        active={false}
        walletType={'hardware'}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when wallet is encrypted software', () => {
    const wrapper = shallow(
      <WalletStatusIndicator
        active={false}
        walletType={'software'}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when wallet is decrypted software', () => {
    const wrapper = shallow(
      <WalletStatusIndicator
        active
        walletType={'software'}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
