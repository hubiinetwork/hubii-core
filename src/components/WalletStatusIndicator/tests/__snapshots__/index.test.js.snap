// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WalletStatusIndicator /> should render correctly when wallet is connected hardware 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WalletStatusIndicator
    active={true}
    walletType="hardware"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "active": true,
      "children": <Tooltip
        arrowPointAtCenter={false}
        autoAdjustOverflow={true}
        mouseEnterDelay={0.1}
        mouseLeaveDelay={0.1}
        placement="right"
        prefixCls="ant-tooltip"
        title="Connected"
        transitionName="zoom-big-fast"
>
        <style__Icon
                src="IMAGE_MOCK"
                walletType="hardware"
        />
</Tooltip>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "arrowPointAtCenter": false,
        "autoAdjustOverflow": true,
        "children": <style__Icon
          src="IMAGE_MOCK"
          walletType="hardware"
/>,
        "mouseEnterDelay": 0.1,
        "mouseLeaveDelay": 0.1,
        "placement": "right",
        "prefixCls": "ant-tooltip",
        "title": "Connected",
        "transitionName": "zoom-big-fast",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "src": "IMAGE_MOCK",
          "walletType": "hardware",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": true,
        "children": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="right"
          prefixCls="ant-tooltip"
          title="Connected"
          transitionName="zoom-big-fast"
>
          <style__Icon
                    src="IMAGE_MOCK"
                    walletType="hardware"
          />
</Tooltip>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrowPointAtCenter": false,
          "autoAdjustOverflow": true,
          "children": <style__Icon
            src="IMAGE_MOCK"
            walletType="hardware"
/>,
          "mouseEnterDelay": 0.1,
          "mouseLeaveDelay": 0.1,
          "placement": "right",
          "prefixCls": "ant-tooltip",
          "title": "Connected",
          "transitionName": "zoom-big-fast",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "src": "IMAGE_MOCK",
            "walletType": "hardware",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<WalletStatusIndicator /> should render correctly when wallet is decrypted software 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WalletStatusIndicator
    active={true}
    walletType="software"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "active": true,
      "children": <Tooltip
        arrowPointAtCenter={false}
        autoAdjustOverflow={true}
        mouseEnterDelay={0.1}
        mouseLeaveDelay={0.1}
        placement="right"
        prefixCls="ant-tooltip"
        title="Unlocked"
        transitionName="zoom-big-fast"
>
        <style__Icon
                src="IMAGE_MOCK"
                walletType="software"
        />
</Tooltip>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "arrowPointAtCenter": false,
        "autoAdjustOverflow": true,
        "children": <style__Icon
          src="IMAGE_MOCK"
          walletType="software"
/>,
        "mouseEnterDelay": 0.1,
        "mouseLeaveDelay": 0.1,
        "placement": "right",
        "prefixCls": "ant-tooltip",
        "title": "Unlocked",
        "transitionName": "zoom-big-fast",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "src": "IMAGE_MOCK",
          "walletType": "software",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": true,
        "children": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="right"
          prefixCls="ant-tooltip"
          title="Unlocked"
          transitionName="zoom-big-fast"
>
          <style__Icon
                    src="IMAGE_MOCK"
                    walletType="software"
          />
</Tooltip>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrowPointAtCenter": false,
          "autoAdjustOverflow": true,
          "children": <style__Icon
            src="IMAGE_MOCK"
            walletType="software"
/>,
          "mouseEnterDelay": 0.1,
          "mouseLeaveDelay": 0.1,
          "placement": "right",
          "prefixCls": "ant-tooltip",
          "title": "Unlocked",
          "transitionName": "zoom-big-fast",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "src": "IMAGE_MOCK",
            "walletType": "software",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<WalletStatusIndicator /> should render correctly when wallet is disconnected hardware 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WalletStatusIndicator
    active={false}
    walletType="hardware"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "active": false,
      "children": <Tooltip
        arrowPointAtCenter={false}
        autoAdjustOverflow={true}
        mouseEnterDelay={0.1}
        mouseLeaveDelay={0.1}
        placement="right"
        prefixCls="ant-tooltip"
        title="Disconnected"
        transitionName="zoom-big-fast"
>
        <style__Icon
                src="IMAGE_MOCK"
                walletType="hardware"
        />
</Tooltip>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "arrowPointAtCenter": false,
        "autoAdjustOverflow": true,
        "children": <style__Icon
          src="IMAGE_MOCK"
          walletType="hardware"
/>,
        "mouseEnterDelay": 0.1,
        "mouseLeaveDelay": 0.1,
        "placement": "right",
        "prefixCls": "ant-tooltip",
        "title": "Disconnected",
        "transitionName": "zoom-big-fast",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "src": "IMAGE_MOCK",
          "walletType": "hardware",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": false,
        "children": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="right"
          prefixCls="ant-tooltip"
          title="Disconnected"
          transitionName="zoom-big-fast"
>
          <style__Icon
                    src="IMAGE_MOCK"
                    walletType="hardware"
          />
</Tooltip>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrowPointAtCenter": false,
          "autoAdjustOverflow": true,
          "children": <style__Icon
            src="IMAGE_MOCK"
            walletType="hardware"
/>,
          "mouseEnterDelay": 0.1,
          "mouseLeaveDelay": 0.1,
          "placement": "right",
          "prefixCls": "ant-tooltip",
          "title": "Disconnected",
          "transitionName": "zoom-big-fast",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "src": "IMAGE_MOCK",
            "walletType": "hardware",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<WalletStatusIndicator /> should render correctly when wallet is encrypted software 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WalletStatusIndicator
    active={false}
    walletType="software"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "active": false,
      "children": <Tooltip
        arrowPointAtCenter={false}
        autoAdjustOverflow={true}
        mouseEnterDelay={0.1}
        mouseLeaveDelay={0.1}
        placement="right"
        prefixCls="ant-tooltip"
        title="Locked"
        transitionName="zoom-big-fast"
>
        <style__Icon
                src="IMAGE_MOCK"
                walletType="software"
        />
</Tooltip>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "arrowPointAtCenter": false,
        "autoAdjustOverflow": true,
        "children": <style__Icon
          src="IMAGE_MOCK"
          walletType="software"
/>,
        "mouseEnterDelay": 0.1,
        "mouseLeaveDelay": 0.1,
        "placement": "right",
        "prefixCls": "ant-tooltip",
        "title": "Locked",
        "transitionName": "zoom-big-fast",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "src": "IMAGE_MOCK",
          "walletType": "software",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": false,
        "children": <Tooltip
          arrowPointAtCenter={false}
          autoAdjustOverflow={true}
          mouseEnterDelay={0.1}
          mouseLeaveDelay={0.1}
          placement="right"
          prefixCls="ant-tooltip"
          title="Locked"
          transitionName="zoom-big-fast"
>
          <style__Icon
                    src="IMAGE_MOCK"
                    walletType="software"
          />
</Tooltip>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "arrowPointAtCenter": false,
          "autoAdjustOverflow": true,
          "children": <style__Icon
            src="IMAGE_MOCK"
            walletType="software"
/>,
          "mouseEnterDelay": 0.1,
          "mouseLeaveDelay": 0.1,
          "placement": "right",
          "prefixCls": "ant-tooltip",
          "title": "Locked",
          "transitionName": "zoom-big-fast",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "src": "IMAGE_MOCK",
            "walletType": "software",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
